src/ZeroLog/EnumArg.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text.Formatting;
+using ZeroLog.Utils;
+
+namespace ZeroLog
+{
+    [StructLayout(LayoutKind.Sequential)]
+    [SuppressMessage("ReSharper", "FieldCanBeMadeReadOnly.Local")]
+    internal struct EnumArg
+    {
+        private IntPtr _typeHandle;
+        private ulong _value;
+
+        public EnumArg(IntPtr typeHandle, ulong value)
+        {
+            _typeHandle = typeHandle;
+            _value = value;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void AppendTo(StringBuffer stringBuffer)
+        {
+            var enumString = EnumCache.TryGetString(_typeHandle, _value, out var enumRegistered);
```

Comment:
Shouldn't it return bool and the string as out intead?

Reply:
Not really: when `enumRegistered` is true, it doesn't necessarily mean the string won't be null. It's an additional information. The true method output is the string, and if it's null, the out parameter gives additional information about the reason why.

Reply:
In which case the name should probably be `GetString` as the convention around `TryGetXXX` is that it returns a bool.